ogr/factory.py

```diff
@@ -117,4 +117,46 @@ def get_service_class(
     for service, service_kls in mapping.items():
         if service in url:
             return service_kls
+
+    return None
+
+
+def get_service_class(
+    url: str, service_mapping_update: Dict[str, Type[GitService]] = None
+) -> Type[GitService]:
+    """
+    Get the matching service class from the url.
+
+    :param url: str (url of the project, e.g. "https://github.com/packit-service/ogr")
+    :param service_mapping_update: custom mapping from  service url (str) to service class
+    :return: Matched class (subclass of GitService)
+    """
+    service_kls = get_service_class_or_none(
+        url=url, service_mapping_update=service_mapping_update
+    )
+    if service_kls:
+        return service_kls
     raise OgrException("No matching service was found.")
+
+
+def get_instances_from_dict(instances: dict) -> Set[GitService]:
```

Comment:
any strong reason for the set?

Reply:
also docstring please: document the format of instances

Reply:
> also docstring please: document the format of instances

Of course.

Reply:
> any strong reason for the set?

Since the order is not importent. (Also easier testing is the reason.)

If you feel it's not necessary, I can change that to list and remove the `__hash__` implementations.

Reply:
> set

I don't care to be honest, I was just curious.